src/config/config.ts

```diff
@@ -1,193 +1,110 @@
+  basicVars.core.nodeEnv === Environment.Dev ||
+  basicVars.core.nodeEnv === Environment.Test
 const nodeEnv = isDev ? Environment.Dev : Environment.Prod
-const port = parseInt(process.env.PORT, 10) || defaults.app.port
-const sessionSecret = process.env.SESSION_SECRET || defaults.app.sessionSecret
 
-/**
- * OTP Life Span for Login. (Should be in miliseconds, e.g. 1000 * 60 * 15 = 15
- * mins).
- */
-const otpLifeSpan =
-  parseInt(process.env.OTP_LIFE_SPAN, 10) || defaults.login.otpLifeSpan
-
-/**
- * TTL of bounce documents in milliseconds.
- */
-const bounceLifeSpan =
-  parseInt(process.env.BOUNCE_LIFE_SPAN, 10) || defaults.bounce.bounceLifeSpan
-
-/**
- * Number of submissions to top up submissions statistic by
- */
-const submissionsTopUp = parseInt(process.env.SUBMISSIONS_TOP_UP, 10) || 0
-
-/**
- * Content Security Policy reporting
- * HelmetJS reportUri param requires non-empty string, so 'undefined' is
- * declared.
- */
-const cspReportUri = process.env.CSP_REPORT_URI || 'undefined'
-if (!process.env.CSP_REPORT_URI) {
-  configLogger.warn('Content Security Policy reporting is not configured.')
-}
-
-/**
- * Chromium executable for PDF generation
- */
-const chromiumBin = process.env.CHROMIUM_BIN
-if (!chromiumBin) {
-  throw new Error(
-    'Path to Chromium executable missing - please specify in CHROMIUM_BIN environment variable',
-  )
+// Load and validate configuration values which are compulsory only in production
+// If environment variables are not present, an error will be thrown
+// They may still be referenced in development
+let prodOnlyVars
+if (isDev) {
+  prodOnlyVars = convict(prodOnlyVarsSchema).getProperties()
+} else {
+  // Perform validation before accessing ses config
+  prodOnlyVars = convict(prodOnlyVarsSchema)
+    .validate({ allowed: 'strict' })
+    .getProperties()
 }
 
-/**
- * FormSG SDK mode.
- * Needed due to the SDK having multiple modes - staging, production,
- * development, and test, whilst the backend only has 2 modes (prod and dev).
- *
- * The SDK modes are required to be set properly so that the correct public keys
- * are used to sign, encrypt, or decrypt data that is passed into the SDK.
- */
-const formsgSdkMode = process.env.FORMSG_SDK_MODE as PackageMode
-if (
-  !formsgSdkMode ||
-  !['staging', 'production', 'development', 'test'].includes(formsgSdkMode)
-) {
-  throw new Error(
-    'FORMSG_SDK_MODE not found or invalid. Please specify one of: "staging" | "production" | "development" | "test" in the environment variable.',
-  )
+// Construct bucket URLs depending on node environment
+// If in development env, endpoint communicates with localstack, a fully
+// functional local AWS cloud stack for hosting images/logos/attachments.
+// Else, the environment variables to instantiate S3 are used.
+
+const awsEndpoint = isDev
+  ? defaults.aws.endpoint
+  : `https://s3.${basicVars.awsConfig.region}.amazonaws.com` // NOTE NO TRAILING / AT THE END OF THIS URL!
```

Comment:
is it possible to fold this within the convict schema and validate it? maybe something similar to `loadS3BucketUrlSchema`? seems a bit out of place considering we have validation for everything else, and especially now that we have a function to validate trailing slashes

Reply:
Yeah the problem is the `isDev` flag and `awsEndpoint ` being different from development and production. For production, it is also dependent on another env var. 

technically, 

```
const validateBucketUrl = (
  val: string,
  { isDev, hasTrailingSlash }: { isDev: boolean; hasTrailingSlash: boolean },
) => {
  if (!validator.isURL(val, { require_tld: !isDev })) {
    throw new Error('must be a url')
  }

```
already helps validate that the url is valid? so nothing like `amazonaws.com//bucket`

Reply:
![image](https://user-images.githubusercontent.com/10674085/91387160-3d8abf00-e867-11ea-8ce1-e21180c7f4e7.png)
 Actually I'm wrong :/

Reply:
any ideas @mantariksh ?

Reply:
what if...you just call `validateBucketUrl` after defining `awsEndpoint`?
```suggestion
  : `https://s3.${basicVars.awsConfig.region}.amazonaws.com` // NOTE NO TRAILING / AT THE END OF THIS URL!
  // Validate to make sure no one ever adds a trailing slash here
  validateBucketUrl(awsEndpoint, { isDev, hasTrailingSlash: false })
```

Reply:
great idea. done